### PR TITLE
Feature/gym 12  layout and routing

### PR DESCRIPTION
**Pull request by PiotrekBaranowski**
Created routing and layout for application in dark theme. 
Styled also accross additional 2 breakpoints: Phone and Tablet

Pictures for view:
Laptop:
image
Tablet:
image
Phone:
image

Clicking on user_icon opens modal:
Laptop / tablet:
image
Phone:
image

### CODE REVIEW

**Review by przemyslawren — APPROVED** [review]
LGTM!

**Review by Valentine-456 — APPROVED** [review]
Design is fire 🔥Aproved

File: src/app/layout/footer/Footer.scss [hunk 1]
```diff
@@ -0,0 +1,7 @@
+.footer {
+	display: flex;
+	flex-direction: row;
+	align-items: center;
+	padding: 0 12px;
+	font-size: 12px;
```
Comment (PiotrekBaranowski) [inline src/app/layout/footer/Footer.scss:6]: To be changed to variables

File: src/app/layout/footer/Footer.tsx [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { useMemo } from 'react';
+import './Footer.scss';
+
+const Footer = (): JSX.Element => {
+	const currentYear: number = useMemo(() => {
+		return new Date().getFullYear();
+	}, []);
+
+	return <div className='footer'>Copyright © {currentYear} Sourcery graduates</div>;
```
Comment (PiotrekBaranowski) [inline src/app/layout/footer/Footer.tsx:9]: better to use html5 semantic -> <footer>

File: src/app/layout/navigation/navigationItem/NavigationItem.scss [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+		width: 36px;
+		height: 36px;
+		color: $gray-700;
+	}
+	.icon_description {
+		display: flex;
+		justify-content: center;
+		align-items: center;
+		margin-top: 4px;
+		font-weight: 500;
```
Comment (PiotrekBaranowski) [inline src/app/layout/navigation/navigationItem/NavigationItem.scss:33]: should be inherited from variables

File: src/app/layout/navigation/navigationItem/NavigationItem.scss [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+		font-size: $font-size-small;
+		line-height: 14px;
```
Comment (PiotrekBaranowski) [inline src/app/layout/navigation/navigationItem/NavigationItem.scss:35]: should be inherited from variables

File: src/app/layout/navigation/navigationItem/NavigationItem.scss [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+		color: $gray-700;
+		text-align: center;
+		padding: 0 4px;
+	}
+}
+
+.active {
+	.navigation_icon {
+		border-radius: 12px;
```
Comment (PiotrekBaranowski) [inline src/app/layout/navigation/navigationItem/NavigationItem.scss:44]: should be inherited from variables

File: src/app/layout/navigation/navigationItem/NavigationItem.tsx [hunk 6]
```diff
@@ -0,0 +1,26 @@
+import './NavigationItem.scss';
+import { AppRoutes } from '@/types/routes';
+import { NavLink } from 'react-router-dom';
+
+interface NavigationItemProps {
+	children: React.ReactNode;
+	navigateTo: AppRoutes;
+	description: string;
+	style?: React.CSSProperties;
+	className?: React.HTMLAttributes<HTMLDivElement> | string;
```
Comment (PiotrekBaranowski) [inline src/app/layout/navigation/navigationItem/NavigationItem.tsx:10]: should be more speific to className

File: src/app/layout/topSection/userAvatar/UserAvatar.tsx [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+	};
+
+	const handleOptionsClick = (): void => {
+		navigate(AppRoutes.OPTIONS);
+		handleMenuClose();
+	};
+
+	return (
+		<div className='user_avatar'>
+			<Box sx={{ display: 'flex', alignItems: 'center', textAlign: 'center' }}>
```
Comment (PiotrekBaranowski) [inline src/app/layout/topSection/userAvatar/UserAvatar.tsx:40]: Could carry over to scss

File: src/app/layout/navigation/Navigation.tsx [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+import FitnessCenterIcon from '@mui/icons-material/FitnessCenter';
+import { AppRoutes } from '@/types/routes';
+
+const Navigation = () => {
+	return (
+		<nav className='navigation'>
+			<NavigationItem navigateTo={AppRoutes.HOME} description={'Home'}>
+				<HomeIcon />
+			</NavigationItem>
+			<NavigationItem navigateTo={AppRoutes.TRAINING} description={'Trainings'}>
```
Comment (Hubertsci) [inline src/app/layout/navigation/Navigation.tsx:15]: Is this tab is meant to address Plans/Routines?

**Pull request merged**